src/app/components/problems/problem-set/problem-set.component.ts

```diff
@@ -202,25 +183,40 @@ export class ProblemSetComponent implements OnInit {
         this.modalService.open(content, {ariaLabelledBy: 'modal-basic-title', centered: true});
     }
 
-    private generateJSONURI(arg: {
-        filename: string,
-        text: string
-    }) {
-        this.filename = arg.filename;
-        this.jsonUri = this.sanitizer.bypassSecurityTrustUrl("data:text/json;charset=UTF-8," + encodeURIComponent(arg.text));
-    }
-
-    getQuestionJson(question: Question): SafeUrl {
-        return this.sanitizer.bypassSecurityTrustUrl("data:text/json;charset=UTF-8," + encodeURIComponent(JSON.stringify(question)));
-    }
-
     getFileName(id: number): string {
         const timestamp = formatDate(new Date(), 'yyyy/MM/dd_HH:mm:ss', 'en');
         return `question-${id}-${timestamp}.json`;
     }
 
-    downloadQuestion(id: number) : SafeUrl {
-        return this.getQuestionJson(this.retrievedQuestions?.filter(c => c.id === id)[0]);
+    downloadQuestion(id: number): void {
+        this.questionService.getQuestion(id).subscribe(question => {
+            const tempElement = document.createElement('a');
```

Comment:
Do you really have to do this much to download? I think there should be an easier way.